File: 0020-valid-parentheses/0020-valid-parentheses.java
class Solution {
	public boolean isValid(String s) {
		Stack<Character> st=new Stack<>();
		for(char ch:s.toCharArray()){
			if(ch=='['||ch=='{'||ch=='(')
				st.push(ch);
			else if(st.size()>0 && ((ch==']' && st.peek()=='[')
				   ||(ch=='}' && st.peek()=='{')
				   ||(ch==')' && st.peek()=='('))) {
					st.pop();
			}
			else return false;
		}

		return st.isEmpty();
	}
}

File: 0020-valid-parentheses/0020-valid-parentheses.js
/**
 * @param {string} s
 * @return {boolean}
 */
var isValid = function(s) {
    s=s.split("").reverse()
    let stack=[]
    let index=0
    while(s.length){
        let temp=s.pop()

        if(!stack.length) stack.push(temp)
        else if(((stack[stack.length-1]==="(" && temp===")")|| (stack[stack.length-1]==="{"&& temp==="}") || (stack[stack.length-1]==="[" && temp==="]")))
            stack.pop()
        else stack.push(temp)
    }
    console.log(stack)
    return stack.length===0?true: false
};
